homeassistant/components/sensor/zha.py

```diff
@@ -111,3 +131,155 @@ def state(self):
             return 'unknown'
 
         return round(float(self._state) / 100, 1)
+
+
+class GenericBatterySensor(Sensor):
+    """ZHA generic battery sensor."""
+
+    value_attribute = 32
+    battery_sizes = {
+        0: 'No battery',
+        1: 'Built in',
+        2: 'Other',
+        3: 'AA',
+        4: 'AAA',
+        5: 'C',
+        6: 'D',
+        7: 'CR2',
+        8: 'CR123A',
+        255: 'Unknown'
+    }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return '%'
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        result = yield from zha.safe_read(
+            self._endpoint.power,
+            ['battery_size', 'battery_quantity', 'battery_voltage']
+        )
+        self._device_state_attributes['battery_size'] = \
+            self.battery_sizes.get(
+                result.get('battery_size', 255),
+                'Unknown'
+            )
+        self._device_state_attributes['battery_quantity'] = result.get(
+            'battery_quantity', 'Unknown')
+        self._state = result.get('battery_voltage', self._state)
+
+
+class CentraliteBatterySensor(GenericBatterySensor):
```

Comment:
I don't think this should live in Home Assistant.

Reply:
"I'd be OKish with this going in and changing to use the normal BatterySensor when zigpy quirks handling is ready, though."

Will leave here until quirks mode is ready.